.github/workflows/manual-publish.yml

```diff
@@ -31,8 +33,23 @@ jobs:
           ssm_parameter_pairs: '/production/common/releasing/pypi/token = PYPI_AUTH_TOKEN'
 
       - uses: ./.github/actions/build
+        id: build
 
       - uses: ./.github/actions/publish
         with:
           token: ${{env.PYPI_AUTH_TOKEN}}
           dry_run: ${{ inputs.dry_run }}
+
+  release-provenance:
+    needs: [ 'build-publish' ]
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.7.0
+    with:
+      base64-subjects: "${{ needs.build-publish.outputs.package-hashes }}"
+      upload-assets: true
```

Comment:
For this workflow, you could tie this into the dry_run dispatch input. That way you can verify the hashes but you don't upload anything?